### PR TITLE
Rename general user settings to account

### PR DESCRIPTION
**Pull request by dbkr**
Everything 'general' is now elsewhere, so this is ready for its new name.

This still needed a bunch of tests adding, so that's about half the content of this PR.

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Comment by dbkr** [comment]
Sonarclud wants the renamed file to be up to test coverage, which admittedly it ought to be anyway, so I'm going to put this on ice while I sort the thing out, convert it to functional etc.

File: test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx [hunk 1]
```diff
@@ -170,7 +238,7 @@ describe("<GeneralUserSettingsTab />", () => {
             expect(within(screen.getByTestId("mx_AccountPhoneNumbers")).getByLabelText("Loading…")).toBeInTheDocument();
         });
 
-        it("should display 3pid email addresses and phone numbers", async () => {
+        it.skip("should display 3pid email addresses and phone numbers", async () => {
```
Comment (robintown) [inline test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx:241]: Why is this skipped?

File: test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx [hunk 2]
```diff
@@ -302,4 +370,53 @@ describe("<GeneralUserSettingsTab />", () => {
...
+    describe("Password change", () => {
+        beforeEach(() => {
+            mockClient.getCapabilities.mockResolvedValue({
+                "m.change_password": {
+                    enabled: true,
+                },
+            });
+        });
+
+        it("should display adialog if password change succeeded", async () => {
```
Comment (robintown) [inline test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx:383]:
```suggestion
        it("should display a dialog if password change succeeded", async () => {
```

**Review by robintown — COMMENTED** [review]
I assume new icons are to come in a later iteration?

**Review by robintown — APPROVED** [review]

File: test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx (hunk 1 above)
Comment (dbkr) [reply test/components/views/settings/tabs/user/AccountUserSettingsTab-test.tsx]: oops, accidental from debugging tests - good spot, thanks.

**Pull request merged**